### ISSUE TITLE
calcSparseGRM() ERROR: unused argument(KINGformat.out)

### ISSUE THREAD

**Issue by DamienTan** [rounakdey/FastSparseGRM#4]
Hi Rounak,
When I ran calcSparseGRM(), an error occurred:
```
>calcSparseGRM(prefix.in,file.score,file.train,file.seg,no_pcs,num_threads,prefix.out,degree,block.size,max.related.block,KINGformat.out)
Error in calcSparseGRM(prefix.in, file.score, file.train, file.seg, no_pcs,  :
  unused argument (KINGformat.out)
```
The command I used:
`R CMD BATCH --vanilla '--args --prefix.in pruned --prefix.out sparseGRM --file.train step3.unrels --file.score step4_pca.score --file.seg king.seg --num_threads 40 --no_pcs 20 --block.size 5000 --max.related.block 5000 --KINGformat.out FALSE --degree 4' calcSparseGRM_wrapper.R calcSparseGRM.Rout`
It didn't occur when I used the previous version of FastSparseGRM several months ago.
I couldn't solve this problem, hope you can provide me some advice.

**Comment by rounakdey** [comment]
Can you please check what the installed version is (you can get it by running sessionInfo()) ? It should be version 1.01. Please also use the most recent calcSparseGRM_wrapper.R file from the extdata folder. I am finding no such problems on my end. Here's the output if I run your command:


R version 4.0.2 (2020-06-22) -- "Taking Off Again"
Copyright (C) 2020 The R Foundation for Statistical Computing
Platform: x86_64-pc-linux-gnu (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

  Natural language support but running in an English locale

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

> library('optparse')
> options(stringsAsFactors=F)
> 
> option_list <- list(
+   make_option("--prefix.in", type="character", default="",
+     help="Prefix to the input PLINK bed file (it should be accompanied by a fam and a bim file with the same prefix)"),
+   make_option("--file.score", type="character",default="",
+     help="Full path to the file containing the PC scores"), 
+   make_option("--file.train", type="character",default="",
+     help="Full path to the file containing the IDs of the training samples using which the individual specific allele frequencies will be calculated. The first two columns need to represent FID and IID respectively."), 
+   make_option("--file.seg", type="character",default="",
+     help="Full path to ibdseg file from KING"), 
+   make_option("--no_pcs", type="integer", default=0,
+     help="Number of PCs to use"),
+   make_option("--num_threads", type="integer", default=0,
+     help="Number of threads within each core to use (if 0, then automatically calculated)"),
+   make_option("--block.size", type="integer", default=5000,
+     help="SNP block size to load in memory"),
+   make_option("--max.related.block", type="integer", default=5000,
+     help="Maximum allowed size of a related sample block"),
+   make_option("--KINGformat.out", type="logical", default=FALSE,
+     help="Output sparse GRM estimates in KING format"),
+   make_option("--degree", type="integer", default=4,
+     help="Estimate kinships for samples related up to which degree (max 4, default 4)"),
+   make_option("--prefix.out", type="character", default="",
+     help="Prefix to the output files")
+ )
> 
> parser <- OptionParser(usage="%prog [options]", option_list=option_list)
> 
> args <- parse_args(parser, positional_arguments = 0)
> opt <- args$options
> print(opt)
$prefix.in
[1] "pruned"

$file.score
[1] "step4_pca.score"

$file.train
[1] "step3.unrels"

$file.seg
[1] "king.seg"

$no_pcs
[1] 20

$num_threads
[1] 40

$block.size
[1] 5000

$max.related.block
[1] 5000

$KINGformat.out
[1] FALSE

$degree
[1] 4

$prefix.out
[1] "sparseGRM"

$help
[1] FALSE

> 
> file.score <- opt$file.score
> file.train <- opt$file.train
> file.seg<-opt$file.seg
> prefix.in <- opt$prefix.in
> num_threads <- opt$num_threads
> block.size <- opt$block.size
> no_pcs <- opt$no_pcs
> prefix.out <- opt$prefix.out
> degree <- opt$degree
> max.related.block <- opt$max.related.block
> KINGformat.out <- opt$KINGformat.out
> 
> library('FastSparseGRM')
Registered S3 methods overwritten by 'tibble':
  method     from  
  format.tbl pillar
  print.tbl  pillar
> print(sessionInfo())
R version 4.0.2 (2020-06-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /n/sw/intel-cluster-studio-2019/compilers_and_libraries_2019.5.281/linux/mkl/lib/intel64_lin/libmkl_gf_lp64.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] FastSparseGRM_1.01 optparse_1.7.3    

loaded via a namespace (and not attached):
 [1] tidyselect_1.2.0       purrr_0.3.4            splines_4.0.2         
 [4] operator.tools_1.6.3   lattice_0.20-41        vctrs_0.5.1           
 [7] generics_0.0.2         GWASExactHW_1.01       getopt_1.20.3         
[10] stats4_4.0.2           mgcv_1.8-31            utf8_1.1.4            
[13] rlang_1.0.6            pillar_1.8.1           glue_1.4.1            
[16] DBI_1.1.3              BiocGenerics_0.34.0    gdsfmt_1.24.1         
[19] GMMAT_1.3.2            GenomeInfoDbData_1.2.3 foreach_1.5.0         
[22] lifecycle_1.0.3        zlibbioc_1.34.0        Biostrings_2.56.0     
[25] codetools_0.2-16       Biobase_2.48.0         SeqArray_1.28.1       
[28] IRanges_2.22.2         GenomeInfoDb_1.24.2    parallel_4.0.2        
[31] fansi_0.4.1            broom_0.5.6            Rcpp_1.0.7            
[34] backports_1.1.8        S4Vectors_0.26.1       XVector_0.28.0        
[37] formula.tools_1.7.1    dplyr_1.0.10           CompQuadForm_1.4.3    
[40] GenomicRanges_1.40.0   grid_4.0.2             cli_3.6.0             
[43] bitops_1.0-6           tools_4.0.2            magrittr_1.5          
[46] RCurl_1.98-1.2         tibble_3.0.1           mice_3.15.0           
[49] SNPRelate_1.22.0       crayon_1.3.4           tidyr_1.2.1           
[52] pkgconfig_2.0.3        ellipsis_0.3.2         Matrix_1.5-3          
[55] assertthat_0.2.1       logistf_1.24.1         iterators_1.0.12      
[58] R6_2.4.1               SeqVarTools_1.26.0     igraph_1.2.5          
[61] nlme_3.1-148           compiler_4.0.2        
> calcSparseGRM(prefix.in,file.score,file.train,file.seg,no_pcs,num_threads,prefix.out,degree,block.size,max.related.block,KINGformat.out)
[1] "Sparse GRM calculation started at 2023-02-24 03:46:51"
Error in file(file, "rt") : cannot open the connection
Calls: calcSparseGRM -> read.table -> file
In addition: Warning message:
In file(file, "rt") :
  cannot open file 'step4_pca.score': No such file or directory
Execution halted


Of course it fails because I don't have those input files, but the program does go into running the calcSparseGRM function without the error of unused argument.

**Comment by DamienTan** [comment]
Thanks Rounak,
I think I solve the problem.
I checked the `calcSparseGRM.Rout`, and I found
```
other attached packages:
[1] FastSparseGRM_0.0 optparse_1.7.3   
```
I still used the old version even though I installed the new version FastSparseGRM. the package is installed under the default directory `/home/cxm/R/library`. Actually, the command still loads the old package which is under `/usr/local/lib64/R/library`.
I'll update the package immediately. Thanks for you soon reply!
